saleor/plugins/sendgrid/plugin.py

```diff
@@ -0,0 +1,255 @@
+import logging
+from dataclasses import asdict
+
+from django.core.exceptions import ValidationError
+
+from ...core.notify_events import NotifyEventType, UserNotifyEvent
+from ..base_plugin import BasePlugin, ConfigurationTypeField
+from ..error_codes import PluginErrorCode
+from ..models import PluginConfiguration
+from . import SendgridConfiguration
+from .tasks import (
+    send_account_confirmation_email_task,
+    send_account_delete_confirmation_email_task,
+    send_fulfillment_confirmation_email_task,
+    send_fulfillment_update_email_task,
+    send_invoice_email_task,
+    send_order_canceled_email_task,
+    send_order_confirmation_email_task,
+    send_order_confirmed_email_task,
+    send_order_refund_email_task,
+    send_password_reset_email_task,
+    send_payment_confirmation_email_task,
+    send_request_email_change_email_task,
+    send_set_user_password_email_task,
+    send_user_change_email_notification_task,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def get_event_to_task_map():
+    return {
+        UserNotifyEvent.ACCOUNT_CONFIRMATION: send_account_confirmation_email_task,
+        UserNotifyEvent.ACCOUNT_SET_CUSTOMER_PASSWORD: (
+            send_set_user_password_email_task
+        ),
+        UserNotifyEvent.ACCOUNT_DELETE: send_account_delete_confirmation_email_task,
+        UserNotifyEvent.ACCOUNT_CHANGE_EMAIL_CONFIRM: (
+            send_user_change_email_notification_task
+        ),
+        UserNotifyEvent.ACCOUNT_CHANGE_EMAIL_REQUEST: (
+            send_request_email_change_email_task
+        ),
+        UserNotifyEvent.ACCOUNT_PASSWORD_RESET: send_password_reset_email_task,
+        UserNotifyEvent.INVOICE_READY: send_invoice_email_task,
+        UserNotifyEvent.ORDER_CONFIRMATION: send_order_confirmation_email_task,
+        UserNotifyEvent.ORDER_CONFIRMED: send_order_confirmed_email_task,
+        UserNotifyEvent.ORDER_FULFILLMENT_CONFIRMATION: (
+            send_fulfillment_confirmation_email_task
+        ),
+        UserNotifyEvent.ORDER_FULFILLMENT_UPDATE: send_fulfillment_update_email_task,
+        UserNotifyEvent.ORDER_PAYMENT_CONFIRMATION: (
+            send_payment_confirmation_email_task
+        ),
+        UserNotifyEvent.ORDER_CANCELED: send_order_canceled_email_task,
+        UserNotifyEvent.ORDER_REFUND_CONFIRMATION: send_order_refund_email_task,
+    }
+
+
+def get_dynamic_template_map():
```

Comment:
Cannot we join `get_event_to_task_map` and `get_dynamic_template_map` to one mapping?

Reply:
I think we can, but imo by the cost of readability:
```
return {
     UserNotifyEvent.ACCOUNT_CONFIRMATION: ("account_confirmation_template_id", send_account_confirmation_email_task)
     ...
}
```
@IKarbowiak  Do you think that this is the better way?

Reply:
For me, it's a better way, as otherwise, you repeated every NotifyEvent. But the decision is up to you :)

Reply:
Also, does it have to be a function instead of a plain dictionary?